doc/design/upgrade.md

```diff
@@ -1,20 +1,39 @@
+After installing a VCH, a VIC engine admin will need to manage the lifecycle of the VCH. One aspect of this lifecycle management is upgrading and patching. For our purposes, patching and upgrading will be treated the same. 
+
+The VIC engine product will release a new, complete, software download bundle for both patches and upgrades, versus, having a sparse patch or a separate upgrade model.
 
 ## Requirements
 
 ### Scope
 
-While a new version is ready, new version's vic-machine upgrade can upgrade one running or powered off VCH appliance. The containers managed by this VCH will be left in existing version. To have a new version container, user need to create new container from upgraded VCH instance. If existing container is stopped and restarted, the old version should be kept.
+vic-machine can upgrade VCH appliance no matter it's running or not, so vic-machine upgrade cannot rely on services running in VCH appliance.
+
+For the VCH appliance, vic-machine upgrade will detect the version difference between the old VCH and itself, this includes detecting the guestinfo changes between the two versions. Then vic-machine upgrade will migrate the guestinfo metadata, and replace the iso files. If anything incorrect happens during this process, vic-machine upgrade will revert back to old version and status. The VCH will have downtime during upgrade, but should not be broken after this process.
+
+The containers managed by this VCH will be left running the existing version. If the container is running, it will work well if it's not coupled with appliance through port-forwarding. But after VCH is upgraded, the container management might not be fully functional if there is configuration or communication level changes.
+
+To update the containers, especially while there is security patch available, two options are available. 
+
+- First is to upgrade the containers from vic-machine. We can migrate container configuration, replace iso file, and then restart the container process. This update requires container downtime as we cannot update tether without stop container process at this time.
+
+ We could upgrade the containers sequentially or in parallel based on user preference.
+
+- Second, leave it to customer like what docker did. User could easily destroy and recreate same containers through docker commands. The new container will be in new version automatically as container iso file is already updated during the appliance upgrade.
+
+In theory, the first option does not add much value, as both options need downtime for the container. If user could accept container stop, they could recreate that container through their own script easily and flexibily, which could be related to their own business requirement.
+Ideally, we should improve tether to separate tether and container lifecycle, so we can replace iso file and restart tether process transparently. Then it will be helpful to have vic-machine upgrade to update containers.
```

Comment:
> This would only address updating tether, not updating kernel, kernel modules or security issues in the container image.

Yeah, you're right. Removed that section. 

For the security issue in container VM kernel, the suggestion is to recreate container, instead of update, cause they will all introduce container downtime.

The container will be updated only after re-creation, in another word, is immutable.